packages/flutter_test/test/finders_test.dart

```diff
@@ -299,6 +299,50 @@ void main() {
       expect(find.bySemanticsLabel('Foo'), findsOneWidget);
       semanticsHandle.dispose();
     });
+
+    testWidgets('Throws StateError if semantics are not enabled (bySemanticsIdentifier)',
+        (WidgetTester tester) async {
+      expect(() => find.bySemanticsIdentifier('Add'), throwsStateError);
```

Comment:
Can you also test the StateError's message, using the `having()` matcher?

Reply:
Something like this?

```dart
    testWidgets('Throws StateError if semantics are not enabled (bySemanticsIdentifier)',
        (WidgetTester tester) async {
      expect(
        () => find.bySemanticsIdentifier('Add'),
        throwsA(isA<StateError>().having((e) => e.message, 'message', contains('Semantics are not enabled')))
      );
    }, semanticsEnabled: false);
```

Reply:
Copied the test from above, shall I update that as well while I am in here?

```dart
    testWidgets('Throws StateError if semantics are not enabled',
        (WidgetTester tester) async {
      expect(() => find.bySemanticsLabel('Add'), throwsStateError);
    }, semanticsEnabled: false);
```

Reply:
> Something like this?
> 
> ```dart
>     testWidgets('Throws StateError if semantics are not enabled (bySemanticsIdentifier)',
>         (WidgetTester tester) async {
>       expect(
>         () => find.bySemanticsIdentifier('Add'),
>         throwsA(isA<StateError>().having((e) => e.message, 'message', contains('Semantics are not enabled')))
>       );
>     }, semanticsEnabled: false);
> ```

Gone ahead and applied this for now...

Reply:
Yes that is correct